src/code/z_camera.c

```diff
@@ -38,7 +38,22 @@
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CB854.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CB880.s")
+s32 Camera_IsSwimming(Camera* camera) {
+    Actor* focalActor = camera->focalActor;
+
+    if (focalActor == &GET_PLAYER(camera->play)->actor) {
+        if (((Player*)focalActor)->stateFlags3 & 0x8000) {
+            // Swimming as Zora
+            return 999;
+        } else {
+            // Swimming as non-Zora
+            return ((Player*)focalActor)->stateFlags1 & 0x8000000;
+        }
+    } else {
+        // Camera not focused on player
+        return 0;
+    }
+}
```

Comment:
So this function returns either `999`, `0x8000000` or 0? Weird...
Should we make defines for those values?
Do we know if it is always used as a boolean check? Otherwise naming it `Is` may not that much sense

Reply:
If the actual return values are unimportant and are never used outside the scope of the function, I'd argue that forgoing defines for these values would be the better option in this case.

Reply:
For modding purposes, we will eventually want an enum/defines I would say.

Reply:
For reference, this function seems intended to be only used internally to camera

Reply:
I believe modifying the internals of camera is a valid use case for mods. It's small documentation, I know, but any knowledge helps further reverse engineering.

Reply:
There's a follow-up camera PR that will focus on a collection of these types of functions that all do little checks similar to this one, 17 functions in total.

https://github.com/engineer124/mm/blob/46d9714f6da68ff97f236aab8980f7eedee6cfa4/src/code/z_camera.c#L292-L531

I think if a macro/enum were to be introduced for this function, then it should be introduced for all 17 of those internal functions. Since those 17 functions will be PR'd shortly, should I bring this dicussion about enums/macros for these functions to that PR, so we can discuss it through a more bigger picture?

Reply:
Yeah I think it's fine to hold since you have some concrete plans to add this and similar enums.